File: src/plugin/inter/SkillcapeCustomizationInterfacePlugin.java
package plugin.inter;

import org.redrune.game.content.entity.actor.player.skills.SkillCapeCustomizer;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.type.InterfacePlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/31/2017
 */
public class SkillcapeCustomizationInterfacePlugin extends InterfacePlugin {
	
	@Override
	public boolean handle(Player player, int interfaceId, int componentId, int itemId, int slotId, int packetId) {
		if (interfaceId == 20) {
			SkillCapeCustomizer.handleSkillCapeCustomizer(player, componentId);
		}
		return true;
	}
	
	@Override
	public void register() {
		registerInterfacePlugin(20);
	}
}

File: src/main/java/plugin/rsinterface/SkillcapeCustomizationInterfacePlugin.kt
package plugin.rsinterface

import org.redrune.game.content.entity.actor.player.skills.SkillCapeCustomizer
import org.redrune.game.content.plugin.type.InterfacePlugin
import org.redrune.game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/31/2017
 */
class SkillcapeCustomizationInterfacePlugin : InterfacePlugin {
    override fun handle(
        player: Player,
        interfaceId: Int,
        componentId: Int,
        itemId: Int,
        slotId: Int,
        packetId: Int,
    ): Boolean {
        if (interfaceId == 20) {
            SkillCapeCustomizer.handleSkillCapeCustomizer(player, componentId)
        }
        return true
    }

    override fun register() {
        registerInterfacePlugin(20)
    }
}